ql/src/java/org/apache/hadoop/hive/ql/io/parquet/serde/ArrayWritableObjectInspector.java

```diff
@@ -139,8 +160,9 @@ public Object getStructFieldData(final Object data, final StructField fieldRef)
     if (data instanceof ArrayWritable) {
       final ArrayWritable arr = (ArrayWritable) data;
       final StructFieldImpl structField = (StructFieldImpl) fieldRef;
-      if (structField.getIndex() < arr.get().length) {
-        return arr.get()[structField.getIndex()];
+      int index = isRoot ? structField.getIndex() : structField.adjustedIndex;
```

Comment:
I don't think this will work. We change `adjustedIndex` without looking at the `isRoot`, so even root-level inspectors have the adjusted index. But when inspecting data, they should still use the original index.